library/Vanilla/Scheduler/Job/CallbackJob.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Scheduler\Job;
+
+use Vanilla\Scheduler\Job\JobExecutionStatus;
+use Vanilla\Scheduler\Job\JobPriority;
+
+/**
+ * Execute a callback as a job.
+ */
+class CallbackJob implements LocalJobInterface {
+
+    /** @var callable */
+    private $callback;
+
+    /** @var array */
+    private $parameters = [];
```

Comment:
Do I need parameters here or can I just use closures or higher order functions? Consider if your class just took a callback. I could make the following higher order function:

```php
public function callWithArgs(callable $callback, array $args) {
    return function () use ($callback, $args) {
        $callback(..$args);
    }
}
```

Moreover, I could just inline a closure fairly easily to whatever is deferring execution:

```php
new CallbackJob(function () use ($discussionID) {
    $this-processNotifications($discussionID);
});
```

Reply:
The parameters here are available to be passed directly to a function, but they're not required. You can certainly create a `CallbackJob` and specify no parameters.